From: David Bauer <mail@david-bauer.net>
Date: Thu, 11 Feb 2021 19:57:26 +0100
Subject: [PATCH] mtd: spi-nor: add support for Winbond W25Q512

The Winbond W25Q512 is a 512mb SPI-NOR chip. It supports 4K sectors as
well as block protection and Dual-/Quad-read.

Tested on: Ubiquiti UniFi 6 LR

Signed-off-by: David Bauer <mail@david-bauer.net>

Ref: https://patchwork.ozlabs.org/project/linux-mtd/patch/20210213151047.11700-1-mail@david-bauer.net/

--- a/drivers/mtd/spi-nor/spi-nor.c
+++ b/drivers/mtd/spi-nor/spi-nor.c
@@ -2552,6 +2552,9 @@ static const struct flash_info spi_nor_i
 			  .fixups = &w25q256_fixups },
 	{ "w25q256jvm", INFO(0xef7019, 0, 64 * 1024, 512,
 			     SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
+	{ "w25q512jv", INFO(0xef4020, 0, 64 * 1024, 1024,
+			SECT_4K | SPI_NOR_QUAD_READ | SPI_NOR_DUAL_READ |
+			SPI_NOR_HAS_TB | SPI_NOR_HAS_LOCK) },
 	{ "w25m512jv", INFO(0xef7119, 0, 64 * 1024, 1024,
 			SECT_4K | SPI_NOR_QUAD_READ | SPI_NOR_DUAL_READ) },
 
